apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/SpannerGranularAccessTest.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests;
+
+[Collection(nameof(SpannerDatabaseRoleFixture))]
+[CommonTestDiagnostics]
+public class SpannerGranularAccessTest
+{
+    private readonly SpannerDatabaseRoleFixture _fixture;
+
+    public SpannerGranularAccessTest(SpannerDatabaseRoleFixture fixture) =>
+        _fixture = fixture;
+
+    [SkippableFact]
+    public async Task ReadWriteWithDbRoleTest()
+    {
+        Skip.If(_fixture.RunningOnEmulator, "Emulator does not support FGAC (Fine Grained Access Control).");
+        string databaseRole = "reader";
+
+        // Create a database role named 'reader' that just have 'SELECT' (Read) privilege on table.
+        using var connection = _fixture.GetConnection();
+        string databaseRoleCreate = $"CREATE ROLE {databaseRole}";
+        string databaseRoleGrant = $"GRANT SELECT ON TABLE {_fixture.TableName} TO ROLE {databaseRole}";
+        var createCmd = connection.CreateDdlCommand(databaseRoleCreate, databaseRoleGrant);
+        await createCmd.ExecuteNonQueryAsync();
+
+        // Try to read data from the given table using reader role.
+        using var connectionWithDbRole = _fixture.GetConnection(new SpannerConnectionStringBuilder { DatabaseRole = databaseRole, DatabaseName = _fixture.DatabaseName });
```

Comment:
And then you can change this methods for several, like, GetReadOnlyConnection etc.